ios/FluentUI/Bottom Sheet/BottomCommandingController.swift

```diff
@@ -0,0 +1,596 @@
+    }
+
+    private var bottomSheetHeroStackHeight: CGFloat { Constants.heroButtonHeight + Constants.BottomSheet.heroStackBottomMargin + bottomSheetHeroStackTopMargin }
+
+    private var heroCommandWidthConstraints: [NSLayoutConstraint] {
+        heroItems.compactMap { (itemToBindingMap[$0] as? HeroItemBinding)?.widthConstraint }
+    }
+
+    private enum ItemLocation {
+        case heroSet
+        case list
+    }
+
+    private class ItemBinding {
+        let item: CommandingItem
+        let view: UIView
+        let location: ItemLocation
+
+        init(item: CommandingItem, view: UIView, location: ItemLocation) {
+            self.item = item
+            self.view = view
+            self.location = location
+        }
+    }
+
+    private class HeroItemBinding: ItemBinding {
+        let widthConstraint: NSLayoutConstraint
+
+        init(item: CommandingItem, view: UIView, location: ItemLocation, widthConstraint: NSLayoutConstraint) {
+            self.widthConstraint = widthConstraint
+            super.init(item: item, view: view, location: location)
+        }
+    }
+
+    private struct Constants {
+        static let heroButtonHeight: CGFloat = 48
+        static let heroButtonWidth: CGFloat = 96
+
+        static let tableViewIconTintColor: UIColor = Colors.textSecondary
+        static let tableViewBackgroundColor: UIColor = Colors.NavigationBar.background
+
+        struct BottomBar {
+            static let cornerRadius: CGFloat = 14
+            static let backgroundColor: UIColor = Colors.NavigationBar.background
+
+            static let bottomOffset: CGFloat = 10
+            static let heroStackLeadingTrailingMargin: CGFloat = 8
+            static let heroStackTopBottomMargin: CGFloat = 16
+
+            static let moreButtonIcon: UIImage? = UIImage.staticImageNamed("more-24x24")
+
+            // TODO: Localize
+            static let moreButtonTitle: String = "More"
+
+            struct Shadow {
+                static let color: CGColor = UIColor.black.cgColor
+                static let opacity: Float = 0.14
+                static let radius: CGFloat = 8
+            }
+        }
+
+        struct BottomSheet {
+            static let expandedFraction: CGFloat = 0.7 // Probably should be more customizable / based on content
+            static let heroStackBottomMargin: CGFloat = 16
+            static let heroStackExpandableTopMargin: CGFloat = 0
+            static let heroStackNonExpandableTopMargin: CGFloat = 16
+            static let heroStackLeadingTrailingMargin: CGFloat = 8
+        }
+    }
+
+}
+
+extension BottomCommandingController: UITableViewDataSource {
+    public func numberOfSections(in tableView: UITableView) -> Int {
+        return expandedListSections.count
+    }
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        assert(section < expandedListSections.count)
```

Comment:
from the declaration I would guess this has the same badness as assertionFailure in that it seems to by default encode the #file. Would recommend just using preconditionFailure